Code/Framework/AzToolsFramework/Tests/TransformComponent.cpp

```diff
@@ -1063,4 +1066,61 @@ R"DELIMITER(<ObjectStream version="1">
             }
         }
     }
+
+    // Fixture provides a root prefab with Transform component and listens for TransformNotificationBus.
+    class TransformComponentActivationTest
+        : public PrefabTestFixture
+        , public TransformNotificationBus::Handler
+    {
+    protected:
+        void SetUpEditorFixtureImpl() override
+        {
+            PrefabTestFixture::SetUpEditorFixtureImpl();
+
+            CreateRootPrefab();
+        }
+
+        void TearDownEditorFixtureImpl() override
+        {
+            BusDisconnect();
+            
+            PrefabTestFixture::TearDownEditorFixtureImpl();
+        }
+        
+        void OnTransformChanged(const Transform& /*local*/, const Transform& /*world*/) override
+        {
+            m_transformUpdated = true;
+        }
+
+        void MoveEntity(AZ::EntityId entityId)
+        {
+            AzToolsFramework::ScopedUndoBatch undoBatch("Move Entity");
+            TransformBus::Event(entityId, &TransformInterface::SetWorldTranslation, Vector3(1.f, 0.f, 0.f));
+        }
+        
+        bool m_transformUpdated = false;
+    };
+    
+    TEST_F(TransformComponentActivationTest, TransformChanged_OnUndoRedo)
+    {
+        AZ::EntityId entityId = CreateEntityWithPrefab("Entity");
+        MoveEntity(entityId);
+        BusConnect(entityId);
+
+        // verify that undoing/redoing move operations fires TransformChanged event
+        Undo();
+        EXPECT_TRUE(m_transformUpdated);
+        m_transformUpdated = false;
+
+        Redo();
+        EXPECT_TRUE(m_transformUpdated);
+        m_transformUpdated = false;
+
+        // verify that simply activating/deactivating an entity does not fire TransformChanged event
+        Entity* entity = nullptr;
+        ComponentApplicationBus::BroadcastResult(entity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+        entity->Deactivate();
+        entity->Activate();
+        EXPECT_FALSE(m_transformUpdated);
```

Comment:
nit: Might be worth actually moving this to a separate test with a separate name, as we're want to try and avoid testing two things at once in a single test if we can help it (makes diagnosing failures much easier, as with descriptive test names)

Reply:
fixxd

Reply:
Great improvement in the test names! Thanks! 😄